SwiftReflector/CustomSwiftCompiler.cs

```diff
@@ -167,6 +167,33 @@ public Version GetCompilerVersion ()
 			return stm;
 		}
 
+		public XDocument ReflectToXDocument (IEnumerable<string> includeDirectories,
+								 IEnumerable<string> libraryDirectories, string extraArgs,
+		                                                 params string [] moduleNames)
+		{
+			var moduleNameAggregate = moduleNames.Aggregate ((s1, s2) => s1 + s2);
+			var pathName = tempDirectory.UniquePath (moduleNameAggregate, null, "xml");
+
+			includeDirectories = includeDirectories ?? new string [] { tempDirectory.DirectoryPath };
+			libraryDirectories = libraryDirectories ?? new string [] { tempDirectory.DirectoryPath };
+
+			var modulesInLibraries = SwiftModuleFinder.FindModuleNames (libraryDirectories, CompilerInfo.Target);
+
+			List<ISwiftModuleLocation> locations = SwiftModuleFinder.GatherAllReferencedModules (modulesInLibraries,
+													     includeDirectories, CompilerInfo.Target);
+			string output = "";
+			try {
+				output = Reflect (locations.Select (loc => loc.DirectoryPath), libraryDirectories, pathName, extraArgs, moduleNames);
+			} finally {
+				locations.DisposeAll ();
+			}
+			ThrowOnCompilerVersionMismatch (output, moduleNames);
+			using (var stm = new FileStream (output, FileMode.Open, FileAccess.Read, FileShare.ReadWrite)) {
+				return XDocument.Load (stm);
```

Comment:
why not directly call `public static System.Xml.Linq.XDocument Load (string uri);` ?

Reply:
Because I didn't want to assume that a file system path would work for a uri instead of a `file://`?